server/app/models/user.rb

```diff
@@ -64,6 +64,12 @@ def accessible_grids
   end
 
   ##
+  # @param [Grid] grid
+  def has_access?(grid)
+    self.accessible_grids.include?(grid)
```

Comment:
Considering how hot this method is (called on all API requests?), this might be worth optimizing to just query for the specific grid, instead of querying for all grids and then doing a linear search?

Something equivalent to the existing `user.grid_ids.include?(grid.id)` or `accessible_grids.find_by(name: name)` methods, which AFAIK avoid loading all the grids?

```ruby
if master_admin?
  true
else
  self.grids.find(grid.id) # XXX: raises error?
end
```

Needs proper tests, though, because I'm not sure how those relation lookups behave :)